src/phtoolbox/validate.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+"""
+Validate app.tar files before deployment to SOAR.
+
+This module provides functionality to validate the contents of app.tar files,
+specifically checking that app.json contains valid data with no empty top-level values.
+"""
+
+import json
+import sys
+import tarfile
+from pathlib import Path
+from typing import Dict, Any, List
+
+
+def validate_app_json(app_json_data: Dict[str, Any]) -> List[str]:
+    """
+    Validate that app.json has no empty top-level values.
+    
+    Args:
+        app_json_data: The parsed JSON data from app.json
+        
+    Returns:
+        List of validation error messages. Empty list means validation passed.
+    """
+    errors = []
+    
+    for key, value in app_json_data.items():
+        if value is None or value == "" or (isinstance(value, (list, dict)) and len(value) == 0):
+            errors.append(f"Top-level key '{key}' has an empty value")
+    
+    return errors
+
+
+def extract_and_validate_app_json(tar_path: str) -> int:
+    """
+    Extract app.json from app.tar and validate its contents.
+    
+    Args:
+        tar_path: Path to the app.tar file
+        
+    Returns:
+        Exit code: 0 for success, 1 for validation failure
+    """
+    try:
+        with tarfile.open(tar_path, 'r') as tar:
+            # Look for app.json in the tar file, excluding resource fork files
+            app_json_members = [member for member in tar.getmembers() 
+                              if member.name.endswith('app.json') and not member.name.startswith('._')]
+            
+            if not app_json_members:
+                print(f"Error: No app.json found in {tar_path}", file=sys.stderr)
+                return 1
+            
+            # Use the first app.json found (there should typically be only one)
+            app_json_member = app_json_members[0]
+            
+            # Extract and parse the JSON
+            app_json_file = tar.extractfile(app_json_member)
+            if app_json_file is None:
+                print(f"Error: Could not extract app.json from {tar_path}", file=sys.stderr)
+                return 1
+            
+            try:
+                content = app_json_file.read()
+                # Try different encodings
+                for encoding in ['utf-8', 'latin-1', 'cp1252']:
+                    try:
+                        decoded_content = content.decode(encoding)
+                        app_json_data = json.loads(decoded_content)
+                        break
+                    except (UnicodeDecodeError, json.JSONDecodeError):
+                        continue
+                else:
+                    print("Error: Could not decode app.json with any supported encoding", file=sys.stderr)
+                    return 1
```

Comment:
I recommend simplifying this code to only support UTF-8. Per [RFC8259](https://datatracker.ietf.org/doc/html/rfc8259#section-8.1) JSON is expected to be in UTF-8:

> JSON text exchanged between systems that are not part of a closed ecosystem MUST be encoded using UTF-8 [[RFC3629](https://datatracker.ietf.org/doc/html/rfc3629)].

```suggestion
                app_json_data = app_json_file.read().decode('utf-8')
```

Reply:
Done.